src/Hosting/TestHost/src/ClientHandler.cs

```diff
@@ -39,7 +39,7 @@ internal ClientHandler(PathString pathBase, ApplicationWrapper application)
             // PathString.StartsWithSegments that we use below requires the base path to not end in a slash.
             if (pathBase.HasValue && pathBase.Value.EndsWith("/"))
             {
-                pathBase = new PathString(pathBase.Value.Substring(0, pathBase.Value.Length - 1));
+                pathBase = new PathString(pathBase.Value[..^1]);
```

Comment:
When using new syntax please include a comment.
```suggestion
                pathBase = new PathString(pathBase.Value[..^1]); // All but the last character
```